### PR TITLE
Set n_jobs=-1 by default for stacked ensemble components

### PR DESCRIPTION
**Pull request by angela97lin**
Thanks to @freddyaboulton for making the suggestion to set `n_jobs = 1` for `test_stacked_estimator_in_pipeline`; looks like that helps keep the memory usage low enough that we don't crash :)

Ran this several times and all green, so we should be able to merge this!
https://app.circleci.com/pipelines/github/alteryx/evalml?branch=ange_stacked_circleci

### CODE REVIEW

**Comment by angela97lin** [comment]
Wanted to see behavior after some work had been done to decrease memory used by tests. Looks like it now fails less frequently, but still occasionally crashes (likely due to memory issues still?)

**Review by freddyaboulton — APPROVED** [review]
@angela97lin Looks good! I tested locally and the delta between this branch and main is about a couple hundred MB. I think this is ok. The footprint on circle-ci should be similar (maybe 3 gb than what it is locally) but would be good to verify.

stacked_ensemble_njobs-1

latest_main

**Comment by angela97lin** [comment]
@freddyaboulton @dsherry 

Here are the results! It looks like on CircleCI, the footprint is approximately 8GB more 😬 

Also weird to see, but stacking takes more time than main on CircleCI (opposite of Freddy's results locally).

Stacking:

stacking

Main:
main


With that in mind... maybe it's still best to hold off on setting n_jobs=-1? Or should we set all instances where we test for n_jobs=-1 for stacking to 1 instead?

**Comment by freddyaboulton** [comment]
@angela97lin Wow that's nuts! I think we can fix this! But before we do I think we should ask ourselves if setting the default to -1 is worth the work. I'm not sure if we ever had that discussion 😅 

I think it is worth it to do so because it impacts the performance of AutoMLSearch. Right now, `IterativeAlgorithm` will create an emsemble pipeline with the default value of `n_jobs` we set in `StackedEnsembleClassifier`/`StackedEnsembleRegressor`. That being said:
- I don't know if we've ever done a test how much slower an ensemble with `n_jobs=1` is than one with `n_jobs=-1`.
- If we're worried about AutoMLSearch, we can just add an `n_jobs` parameter to `_make_stacked_ensemble_pipeline`. That will make sure the ensemble has the same value for `n_jobs` that the `IterativeAlgorithm` has. If we do this, we can keep the default value as-is and not impact AutoMLSearch. 

If we want to change the default to `-1` and not crash circleci I think this is how we do it:

- The problem is that now every test in `test_stacked_ensemble_classifier` and `test_stacked_ensemble_regressor` uses `n_jobs=-1` for the ensemble (and I noticed some of the pipelines too, e.g. `test_stacked_ensemble_multilevel`).  I think if we make sure `n_jobs=1` for every ensemble test except 1, we'll be in better shape. 
- I'm shocked the memory on circleci is that much worse. Can you try the same experiment locally to make sure I didn't just mess something up?

**Comment by angela97lin** [comment]
Tried locally, and got similar memory usage. I'll try running on CircleCI again 🤔 

Stacking (this PR):
STACKING_LOCAL

Main:
MAIN_LOCAL

**Comment by angela97lin** [comment]
Results from CircleCI:

Stacking:
stacking

Main:
main

Looks like it still takes up an extraordinary amount more memory than main on CircleCI even though they're essentially equivalent locally. 😕 

Going to try some of @freddyaboulton's suggestions on setting all but maybe one of the tests to use n_jobs=1, as I agree that there certainly is value to this change!

**Comment by angela97lin** [comment]
@dsherry @freddyaboulton 

I updated some more tests, and ran a memory profiler multiple times (each time on a new SSH session since I noticed using the same one caused weird behavior). Here are the results:

Main:

main_new


Stacking:
stacking_diff_diff
stacking_diff
stacking_new_new
stacking_final

It looks like it varies quite some bit between each trial, but it's certainly doesn't take much more memory than main--in one trial it even took less 🤔 .

In that case, I think we can be all set with merging this and not worrying about memory issues that might happen. 😁

**Review by freddyaboulton — APPROVED** [review]
@angela97lin Thank you for testing this! I think this is good to merge. I believe all the other places where ensembling happens (e.g. automl unit tests) mock fit and score. Might be good to verify that though hehe.

File: evalml/tests/component_tests/test_components.py [hunk 1]
```diff
@@ -845,9 +845,9 @@ def test_serialization(X_y_binary, tmpdir, helper_functions):
             component = helper_functions.safe_init_component_with_njobs_1(component_class)
         except EnsembleMissingPipelinesError:
             if (component_class == StackedEnsembleClassifier):
-                component = component_class(input_pipelines=[make_pipeline_from_components([RandomForestClassifier()], ProblemTypes.BINARY)])
+                component = component_class(input_pipelines=[make_pipeline_from_components([RandomForestClassifier()], ProblemTypes.BINARY)], n_jobs=1)
```
[the remark this answers is not in the document]
Comment (angela97lin) [reply evalml/tests/component_tests/test_components.py:848]: Oo, that's true! Will do 😁

**Pull request merged**